ion/binaryreader.go

```diff
@@ -36,6 +37,36 @@ func newBinaryReaderBuf(in *bufio.Reader, cat Catalog) Reader {
 	return r
 }
 
+// Reset causes the binary reader to start reading from the given input bytes
+// while skipping most of the initialization steps needed to prepare the
+// reader. Reset is most commonly called with the same bytes as the reader
+// was originally created with (e.g. via NewReaderBytes) as an optimization
+// when the same data needs to be read multiple times.
+//
+// While it is possible to call Reset with different input bytes, the Reader
+// will only work correctly if the new bytes contain the exact same binary
+// version marker and local symbols as the original input. If there are any
+// doubts whether this is the case, it is instead recommended to create a
+// new Reader using NewReaderBytes (or NewReaderCat) instead. Attempting to
+// reuse a binaryReader with inconsistent input bytes will cause the reader
+// to return errors, misappropriate values to unrelated or non-existent
+// attributes, or incorrectly parse data values.
+func (r *binaryReader) Reset(in []byte) error {
```

Comment:
If you don't mind, can you add a note about the visibility of this function for those of us who aren't using Go on a regular basis?

Reply:
The function is exported, so anyone can call it. It's not included in the `Reader` interface, which is the return type from the `NewReader*` functions. In order to use this, one needs to define their own Go interface similar to this:

```go
type resettableReader interface {
	ion.Reader
	Reset([]byte) error
}
```

and perform a type assertion on the interface in order to use it. For example:

```go
func resetOrNew(reader ion.Reader, content []byte) ion.Reader { 
	if rr, ok := reader.(resettableReader); ok {
		_ = rr.Reset(content)
		return rr
	}
	return ion.NewReaderBytes(content)
}
```